ztp/resource-generator/src/watcher.py

```diff
@@ -56,30 +56,34 @@ def watch_sites(self, rv):
 
 
 class PolicyGenWrapper(Logger):
-    def __init__(self, paths: list):
+    def __init__(self, paths: list, resourcename: str = 'siteconfigs'):
         try:
-            folders = [{'input': paths[0], 'output': paths[1]}]
-            src = '/usr/src/hook/cnf-features-deploy'
-            dest = '/tmp/cnf-features-deploy'
+            # Copy the ztp dir to /tmp to allow non-root file creation
+            src = '/usr/src/hook/ztp'
+            dest = '/tmp/ztp'
+            shutil.rmtree(dest, ignore_errors=True)
             shutil.copytree(src, dest)
-            cwd = '/tmp/cnf-features-deploy/ztp/ztp-policy-generator'
-            command = 'kustomize build --enable-alpha-plugins'
-            oneliner_file = 'policyGenerator.yaml'
+            if resourcename == 'siteconfigs':
+                template_dir = '/tmp/ztp/source-cluster-crs'
+            elif  resourcename == 'policygentemplates':
+                template_dir = '/tmp/ztp/source-policy-crs'
+            else:
+                raise Exception('Unsupported resource name')
+            cwd = os.path.join(
+                '/tmp/ztp/ztp-policy-generator',
+                'kustomize/plugin/policyGenerator/v1/policygenerator/')
+            command = [
+                './PolicyGenerator',
+                'dummy_arg',
+                paths[0],
+                template_dir,
+                paths[1],
+                'true', 'false', 'true']
```

Comment:
These booleans change between siteconfigs and policygentemplates. For policygentemplates it should be "true false false" and for siteconfigs it should be  true false true